packages/core/src/asciidoctor-core-api.js

```diff
@@ -394,7 +394,7 @@ AbstractBlock.prototype.getSourceLocation = function () {
  * @memberof AbstractBlock
  */
 AbstractBlock.prototype.getCaption = function () {
-  return this.$caption()
+  return this.$caption() === Opal.nil ? undefined : this.$caption()
```

Comment:
Could you please assign the result of the function `this.$caption()`? Both for consistency and probably performance as the `$caption` method is [not a simple accessor](https://github.com/asciidoctor/asciidoctor/blob/194d191f24dd6a069055c903cd7019772cfa988f/lib/asciidoctor/abstract_block.rb#L244).

```suggestion
  var caption = this.$caption()
  return caption === Opal.nil ? undefined : caption
```